### PR TITLE
added replaceAtIndex action

### PR DESCRIPTION
**Pull request by lucasfeliciano**
related to #5

### CODE REVIEW

File: reducers/index.js [hunk 1]
```diff
@@ -37,6 +37,15 @@ module.exports = function navigate(state, {type, payload}) {
         stack: stack.splice(index, 1, payload),
       });
 
+    case REPLACE_AT_INDEX:
+      if (payload.index > stack.count() - 1) {
+        return state;
```
Comment (Kureev) [inline reducers/index.js]: we need to throw there.

Comment (lucasfeliciano) [inline reducers/index.js]: done ;)

File: test/reducers/navigator.spec.js [hunk 2]
```diff
@@ -56,4 +56,18 @@ describe('navigate reducer', () => {
     expect(state.stack.count()).to.be.equal(2);
     expect(state.stack.first()).to.be.equal('test');
   });
+
+  it('expect REPLACE_AT_INDEX to replace the route at the index passed', () => {
```
Comment (Kureev) [inline test/reducers/navigator.spec.js:60]: add test for a negative scenario (when you trying to replace an index which is out of boundaries)

Comment (lucasfeliciano) [inline test/reducers/navigator.spec.js:60]: Dont you think this is unnecessary, because if this happens it will throw and the test will not pass.

File: reducers/index.js [hunk 3]
```diff
@@ -37,6 +39,16 @@ module.exports = function navigate(state, {type, payload}) {
         stack: stack.splice(index, 1, payload),
       });
 
+    case REPLACE_AT_INDEX:
+      invariant(
+        payload.index < stack.count() - 1 && payload.index > 0,
+        'Index out of bounds when trying to replaceAtIndex',
```
Comment (Kureev) [inline reducers/index.js]: Sometimes it's handy to know which values you actually have. So, for instance, this message can be enhanced:

```
Index out of bounds when trying to replaceAtIndex: your navigation stack contains ${stack.count()} elements, but you're trying to replace an element with index ${payload.index}
```

File: test/reducers/navigator.spec.js (hunk 2 above)
Comment (Kureev) [inline test/reducers/navigator.spec.js:60]: If at some point we'll remove an invariant (for some reason) or it'll be a broken (always return `true`, for instance) - test will show it.

Comment (lucasfeliciano) [inline test/reducers/navigator.spec.js:60]: :+1:  on it

Comment (lucasfeliciano) [inline test/reducers/navigator.spec.js:60]: done

File: reducers/index.js (hunk 3 above)
Comment (lucasfeliciano) [inline reducers/index.js]: done

File: reducers/index.js [hunk 4]
```diff
@@ -37,6 +39,18 @@ module.exports = function navigate(state, {type, payload}) {
         stack: stack.splice(index, 1, payload),
       });
 
+    case REPLACE_AT_INDEX:
+      invariant(
+        payload.index < stack.count() - 1 && payload.index > 0,
+        'Index out of bounds when trying to replaceAtIndex: ' +
+        'your navigation stack contains ${stack.count()} elements, ' +
+        'but you\'re trying to replace an element with index ${payload.index}',
```
Comment (Kureev) [inline reducers/index.js]: `${}` works only if you wrap it into `

Comment (lucasfeliciano) [inline reducers/index.js]: Damn copy and paste :P

Comment (lucasfeliciano) [inline reducers/index.js]: done
